File: Week_03-递归-分治-回溯/70.爬楼梯.cpp
/*
 * @lc app=leetcode.cn id=70 lang=cpp
 *
 * [70] 爬楼梯
 */

// @lc code=start
class Solution {
public:

int climbStairs(int n) {

#if 1
//动态规划，从a->z，那么一步步来，先到b、c、d...
//大问题分解为小问题，并且利用小问题得到的中间结果
     vector<int>steps(n+1,0);
     steps[0]=1;
     steps[1]=1;
     for(int i=2;i<=n;i++)
     {
         steps[i]=steps[i-2]+steps[i-1];
     }
     return steps[n];
}
#endif
};
// @lc code=end

File: Week_03-递归-分治-回溯/70.爬楼梯.py
#
# @lc app=leetcode.cn id=70 lang=python
#
# [70] 爬楼梯
#

# @lc code=start
class Solution(object):
    def climbStairs(self, n):
        """
        :type n: int
        :rtype: int
        """
        step = [0]*(n+1)
        step[0] = 1
        step[1] = 1
        for i in range(2, n+1):
            step[i] = step[i-1] + step[i-2]
        return step[n]
# @lc code=end

